pkg/system/resources.go

```diff
@@ -1,76 +1,73 @@
 package system
 
 import (
-	"os"
+	"fmt"
+	"strings"
 
 	corev1 "k8s.io/api/core/v1"
+
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
-// Values will likely need to be tweaked as we get more usage data. They are currently based
-// on metrics we have collected from internal use. You can override these values by setting
-// the corresponding environment variable.
-var (
-	mi = int64(1 << 20) // 1 MiB in bytes
-	gi = int64(1 << 30) // 1 GiB in bytes.
-
-	registryMemoryRequest = *resource.NewQuantity(128*mi, resource.BinarySI)    // REGISTRY_MEMORY_REQUEST
-	registryMemoryLimit   = *resource.NewQuantity(512*mi, resource.BinarySI)    // REGISTRY_MEMORY_LIMIT
-	registryCPURequest    = *resource.NewMilliQuantity(200, resource.DecimalSI) // REGISTRY_CPU_REQUEST
+var ErrInvalidResourceSpecification = fmt.Errorf("invalid resource specification")
 
-	buildkitdMemoryRequest = *resource.NewQuantity(256*mi, resource.BinarySI)    // BUILDKITD_MEMORY_REQUEST
-	buildkitdMemoryLimit   = *resource.NewQuantity(10*gi, resource.BinarySI)     // BUILDKITD_MEMORY_LIMIT
-	buildkitdCPURequest    = *resource.NewMilliQuantity(800, resource.DecimalSI) // BUILDKITD_CPU_REQUEST
+// ValidateResources is used by the CLI to validate that the values passed
+// can be parsed as a resource.Quantity.
+func ValidateResources(resources ...string) error {
+	for _, rs := range resources {
+		if rs == "" {
+			continue
+		}
 
-	buildkitdServiceMemoryRequest = *resource.NewQuantity(128*mi, resource.BinarySI)    // BUILDKITD_SERVICE_MEMORY_REQUEST
-	buildkitdServiceMemoryLimit   = *resource.NewQuantity(256*mi, resource.BinarySI)    // BUILDKITD_SERVICE_MEMORY_LIMIT
-	buildkitdServiceCPURequest    = *resource.NewMilliQuantity(200, resource.DecimalSI) // BUILDKITD_SERVICE_CPU_REQUEST
-)
+		req, limit, both := strings.Cut(rs, ":")
+		if !both {
+			_, err := resource.ParseQuantity(rs)
+			return err
+		}
 
-func RegistryResources() corev1.ResourceRequirements {
-	return corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("REGISTRY_MEMORY_REQUEST", registryMemoryRequest),
-			corev1.ResourceCPU:    envOrDefault("REGISTRY_CPU_REQUEST", registryCPURequest),
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("REGISTRY_MEMORY_LIMIT", registryMemoryLimit),
-		},
+		_, err := resource.ParseQuantity(req)
+		if err != nil {
+			return err
+		}
+		_, err = resource.ParseQuantity(limit)
+		if err != nil {
+			return err
+		}
```

Comment:
Is it important to validate that limit is greater than request?

Reply:
I agree. I updated this to now check that the request is not greater than the limit.